networkx/algorithms/isomorphism/isomorph.py

```diff
@@ -17,47 +17,68 @@
 
 
 @nx._dispatchable(graphs={"G1": 0, "G2": 1})
-def could_be_isomorphic(G1, G2):
+def could_be_isomorphic(G1, G2, *, properties="dtc"):
     """Returns False if graphs are definitely not isomorphic.
     True does NOT guarantee isomorphism.
 
     Parameters
     ----------
     G1, G2 : graphs
-       The two graphs G1 and G2 must be the same type.
+       The two graphs `G1` and `G2` must be the same type.
+
+    properties : str, default="dct"
+       Determines which properties of the graph are checked. Each character
+       indicates a particular property as follows:
+
+       - if ``"d"`` in `properties`: degree of each node
+       - if ``"t"`` in `properties`: number of triangles for each node
+       - if ``"c"`` in `properties`: number of maximal cliques for each node
+
+       Unrecognized characters are ignored. The default is ``"dtc"``, which
+       compares the sequence of ``(degree, num_triangles, num_cliques)`` properties
+       between `G1` and `G2`.
+
+    Returns
+    -------
+    bool
+       A Boolean value representing whether `G1` could be isomorphic with `G2`
+       according to the specified `properties`.
 
     Notes
     -----
-    Checks for matching degree, triangle, and number of cliques sequences.
     The triangle sequence contains the number of triangles each node is part of.
     The clique sequence contains for each node the number of maximal cliques
     involving that node.
-
     """
 
     # Check global properties
     if G1.order() != G2.order():
         return False
 
-    # Check local properties
-    d1 = G1.degree()
-    t1 = nx.triangles(G1)
-    clqs_1 = list(nx.find_cliques(G1))
-    c1 = Counter(chain.from_iterable(nx.find_cliques(G1)))  # number of cliques
-    props1 = [[d, t1[v], c1[v]] for v, d in d1]
-    props1.sort()
-
-    d2 = G2.degree()
-    t2 = nx.triangles(G2)
-    clqs_2 = list(nx.find_cliques(G2))
-    c2 = Counter(chain.from_iterable(nx.find_cliques(G2)))  # number of cliques
-    props2 = [[d, t2[v], c2[v]] for v, d in d2]
-    props2.sort()
-
-    if props1 != props2:
+    properties_to_check = set(properties)
+    G1_props, G2_props = [], []
+
+    # Degree sequence
+    if "d" in properties_to_check:
+        G1_props.append(G1.degree())
+        G2_props.append(G2.degree())
```

Comment:
Some quick timing suggest that computing the properties dominates the time over the sort/compare part. If so, we could add a shortcut check after "d" and "t": build the table-so-far and sort/compare it. That would mean the sort/compare times would happen more than once. But we'd get a faster `False` due to avoiding the slower properties. Thoughts?

Reply:
I think that makes sense - so if I'm understanding correctly it's sort of a combination of both approaches. Something like:
 1. Compute the property
 2. Compare the property itself *independently* between the two graphs. If not equal, `return False`
 3. If equal, then the property is appended to the "full property list" for the more specific check at the end.

I agree this makes sense if the computation of the properties themselves is the bottleneck, rather than the collating/sorting of the properties once they're computed. I'd expect that to be the case but never actually bothered to check!

Reply:
I ran %timeit for different sized Graphs and it was orders of magnitude longer for each step up the degree/triangle/clique hill... And the table-prep and sorting and comparing took about the same as degree. So it does seem to be how we'd expect. :)

Reply:
If that's the case (which seems to be as expected), might as well compare the "full property list" after each property that is computed--why wait until the very end to do so?

Reply:
Done as of 6986d68 ! I think it's also worth adding a comment to explain why things are organized this way (haven't done this yet...)

Reply:
> (haven't done this yet...)

Comment added in 613c81d !